build/AssemblyRedirectionSourceGenerator.cs

```diff
@@ -31,13 +31,20 @@ public static void Generate(string assembliesFolderPath, string generatedFilePat
 
         var sourceContents = GenerateSourceContents(assemblies);
 
-        File.WriteAllText(generatedFilePath, sourceContents, Encoding.UTF8);
```

Comment:
I assume we don't care about BOM here ([docs](https://learn.microsoft.com/en-us/dotnet/api/system.io.file.writealltext?view=net-7.0#system-io-file-writealltext(system-string-system-string)))